plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/monitor/utils/FlexibleEngineMonitorConverter.java

```diff
@@ -186,11 +170,10 @@ public StringBuilder buildProjectQueryUrl(String region) {
 
     private Map<String, String> getFlexibleEngineMonitorParams(
             ResourceMetricRequest resourceMetricRequest,
-            FlexibleEngineNameSpaceKind flexibleEngineNameSpaceKind) {
+            MetricInfoList metricInfo) {
         Map<String, String> params = new HashMap<>();
-        params.put("namespace", FlexibleEngineNameSpaceKind.ECS_SYS.toValue());
-        params.put("metric_name", getMetricName(resourceMetricRequest.getMonitorResourceType(),
-                flexibleEngineNameSpaceKind));
+        params.put("namespace", metricInfo.getNamespace());
+        params.put("metric_name", metricInfo.getMetricName());
         params.put("from", String.valueOf(resourceMetricRequest.getFrom()));
         params.put("to", String.valueOf(resourceMetricRequest.getTo()));
         params.put("period", String.valueOf(resourceMetricRequest.getPeriod()));
```

Comment:
should we add null checks for adding query parameters? because not all parameters are manadatory.

Reply:
Ok.